src/read_utils.jl

```diff
@@ -2,6 +2,44 @@ using DelimitedFiles
 
 # ---------------------------------------------------------------------------------------- #
 
+struct bandstructure
```

Comment:
Some comments:

1. The usual convention in Julia is to CamelCase type names. I.e. this should be `Bandstructure` or `BandStructure`.
   In this case, I think something like `BandSummary` might be better though.
2. We should nearly always avoid using abstract types for fields. With abstract types, the compiler cannot know in advance what the type of the field is - so everything using this type would become type unstable (and all the data would have to be "boxed", i.e. instead of a field being stored directly, the field ends up being just a pointer to the actual data). So the fields here should be e.g. `:: String` and `:: Vector{Int}`.
    (If one needs to keep the option for generic field types, one can use parametric types).
3. Might be nice to store the irrep labels (as `irlabs`), so we can print out the actual symmetry vector nicely.